CoreRemoting/Channels/Tcp/TcpClientChannel.cs

```diff
@@ -54,11 +57,17 @@ public void Connect()
 
         _tcpClient.Events.ExceptionEncountered += OnError;
         _tcpClient.Events.MessageReceived += OnMessage;
+        _tcpClient.Events.ServerDisconnected += OnDisconnected;
         _tcpClient.Connect();
 
         _tcpClient.Send(new byte[1] { 0x0 }, _handshakeMetadata);
     }
 
+    private void OnDisconnected(object o, DisconnectionEventArgs disconnectionEventArgs)
+    {
+        Disconnected.Invoke();
```

Comment:
Should be Disconnected?.Invoke() to support calling without subscribers

Reply:
I'm agree